src/main/java/chatbot/impl/TaskStorageImpl.java

```diff
@@ -0,0 +1,142 @@
+package chatbot.impl;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import chatbot.Task;
+import chatbot.TaskStorage;
+import chatbot.impl.tasks.AbstractTask;
+
+/**
+ * Implements the TaskStorage interface to manage tasks in a custom file-based storage system.
+ */
+public class TaskStorageImpl implements TaskStorage {
+
+    private static final Path FILE_PATH = Paths.get("./data/tasks.txt");
+
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Constructs a new TaskStorageImpl.
+     * Initializes the task list and loads existing tasks from the file.
+     */
+    public TaskStorageImpl() {
+        tasks = new ArrayList<>();
+
+        try {
+            Files.createDirectories(Paths.get("./data"));
+            if (!Files.exists(FILE_PATH)) {
+                Files.createFile(FILE_PATH);
+            }
+
+            List<String> lines = Files.readAllLines(FILE_PATH);
+            for (String line : lines) {
+                tasks.add(AbstractTask.deserialize(line));
+            }
+        } catch (IOException e) {
+            // Todo: Proper exception handling
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation adds the task to the in-memory list and saves it to the file.
+     */
+    @Override
+    public void addTask(Task task) {
+        tasks.add(task);
+        saveTasks();
+    }
+
+    @Override
+    public Task getTask(int taskIdx) {
+        return tasks.get(taskIdx);
+    }
+
+    @Override
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation marks the task as done in the in-memory list and saves the changes.
+     */
+    @Override
+    public void setTaskAsDone(int taskIdx) {
+        tasks.get(taskIdx).setDone(true);
+        saveTasks();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation marks the task as not done in the in-memory list and saves the changes.
+     */
+    @Override
+    public void setTaskAsNotDone(int taskIdx) {
+        tasks.get(taskIdx).setDone(false);
+        saveTasks();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation removes the task from the in-memory list and saves the changes.
+     */
+    @Override
+    public void deleteTask(int taskIdx) {
+        tasks.remove(taskIdx);
+        saveTasks();
+    }
+
+    /**
+     * Returns a string representation of all tasks in the storage.
```

Comment:
Seems like this JavaDoc comment is in front of the wrong method (should be in front of toString()?)